plugins/catalog/src/CatalogClientWrapper.ts

```diff
@@ -68,6 +68,19 @@ export class CatalogClientWrapper implements CatalogApi {
     });
   }
 
+  async getAttachment(name: EntityName, key: string): Promise<Blob> {
+    // TODO: Here we could set the header, but I think most of the use cases
+    // won't use the function, e.g. if I embed the url into an image like
+    // <img src={getAttachmentUrl(...)} ...
+    return await this.client.getAttachment(name, key);
+  }
+
+  async getAttachmentUrl(name: EntityName, key: string): Promise<string> {
+    // TODO: Auth headers doesn't work well together with these urls...
+    // cookies would be better here. For now we skip this topic completly.
+    return await this.client.getAttachmentUrl(name, key);
+  }
```

Comment:
This is problematic, but only if you use this approach of auth. If you use a cookie based reverse proxy in front of Backstage, this isn't a problem.

Reply:
In this case we could also include the token in a query parameter, but that isn't nice either...

Reply:
Btw, I found a solution here that isn't very nice, but at least doesn't break... We now do a request and return a `data:` URI here.